src/Agent.Listener/CommandSettings.cs

```diff
@@ -127,6 +127,39 @@ public bool GetRunAsService()
                 defaultValue: false);
         }
 
+        public bool GetEnableAutoLogon()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.EnableAutoLogon,
+                description: StringUtil.Loc("EnableAutoLogon"),
+                defaultValue: !Unattended);
+        }
+
+        public bool GetRestartPermission()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.RestartIfNeeded,
+                description: StringUtil.Loc("RestartNow"),
+                defaultValue: true);
+        }
+
+        public bool GetContinueWithoutAutoLogon()
+        {
+            return TestFlagOrPrompt(
+                name: Constants.Agent.CommandLine.Flags.ContinueWithoutAutoLogon,
+                description: StringUtil.Loc("ContinueWithoutAutoLogon"),
+                defaultValue: false);
```

Comment:
This seems to be unnecessary. If user opts for `Enable Auto logon`, it should enable or error out.
If user didn't opt for it, dont enable.